code/datums/components/pellet_cloud.dm

```diff
@@ -0,0 +1,121 @@
+/datum/component/pellet_cloud
+	var/projectile_type
+	var/num_pellets
+	var/list/pellets = list()
+	var/terminated = 0
+	var/hits = 0
+	var/list/targets_hit = list()
+	var/atom/aim_target
+	var/proj_name
+	var/radius = 4
+	var/holdup = FALSE
+
+/datum/component/pellet_cloud/Initialize(projectile_type, magnitude=5)
+	if(!istype(parent, /obj/item/ammo_casing) && !isgrenade(parent))
+		return COMPONENT_INCOMPATIBLE
+	if(isgrenade(parent))
+		holdup = TRUE
+
+	src.projectile_type = projectile_type
+	if(istype(parent, /obj/item/ammo_casing))
+		num_pellets = magnitude
+	else if(isgrenade(parent))
+		radius = magnitude
+
+	var/obj/projectile/p = projectile_type
+	proj_name = initial(p.name)
+
+/datum/component/pellet_cloud/Destroy(force, silent)
+	if(holdup)
+		return QDEL_HINT_LETMELIVE
+	. = ..()
+
+/datum/component/pellet_cloud/RegisterWithParent()
+	if(istype(parent, /obj/item/ammo_casing))
+		RegisterSignal(parent, COMSIG_PELLET_CLOUD_INIT, .proc/create_pellets)
+	else if(isgrenade(parent))
+		RegisterSignal(parent, COMSIG_GRENADE_PRIME, .proc/circle_pellets)
+
+/datum/component/pellet_cloud/UnregisterFromParent()
+	if(holdup)
+		return QDEL_HINT_LETMELIVE
+	UnregisterSignal(parent, list(COMSIG_PELLET_CLOUD_INIT, COMSIG_GRENADE_PRIME))
+
+//datum/component/pellet_cloud/proc/projectile_hit(atom/fired_from, atom/movable/firer, atom/target, Angle)
+	//do_shrapnel(firer, target)
+
+/datum/component/pellet_cloud/proc/create_pellets(obj/item/ammo_casing/A, target, user, fired_from)
+	aim_target = target
+	for(var/pelleties = 0, pelleties < num_pellets, pelleties++)
```

Comment:
```suggestion
	for(var/i in 1 to num_pellets)
```
You'll want the C loop for cases where you want to manipulate the control variable and have it affect how many times the proc is run.
If you don't, the list loop skips extra checks and just does what is intended.